src/categorical_algebra/CSetDataStructures.jl

```diff
@@ -671,64 +656,107 @@ function Base.show(io::IO, acs::T) where {S,T<:StructACSet{S}}
 end
 
-parent(row::StructACSetRow) = getfield(row, :parent)
-idx(row::StructACSetRow) = getfield(row, :idx)
+parent(row::ACSetRow) = getfield(row, :parent)
+idx(row::ACSetRow) = getfield(row, :idx)
 
 # - Tables.jl interface
 
-Tables.istable(sat::StructACSetTable) = true
+Tables.istable(sat::ACSetTable) = true
 
-Tables.columnaccess(sat::StructACSetTable) = true
-Tables.columns(sat::StructACSetTable) = sat
+Tables.columnaccess(sat::ACSetTable) = true
+Tables.columns(sat::ACSetTable) = sat
 
-Tables.getcolumn(sat::StructACSetTable{T,ob}, i::Int) where {T,ob} =
+Tables.getcolumn(sat::ACSetTable{T,ob}, i::Int) where {T,ob} =
   Base.getproperty(sat, outgoing(parent(sat), ob)[i])
 
-Base.getproperty(sat::StructACSetTable, nm::Symbol) = subpart(parent(sat), :, nm)
+Base.getproperty(sat::ACSetTable, nm::Symbol) = subpart(parent(sat), :, nm)
 
-Tables.getcolumn(sat::StructACSetTable, nm::Symbol) = Base.getproperty(sat, nm)
+Tables.getcolumn(sat::ACSetTable, nm::Symbol) = Base.getproperty(sat, nm)
 
-Base.propertynames(sat::StructACSetTable{T,ob}) where {T,ob} = outgoing(parent(sat), ob)
+Base.propertynames(sat::ACSetTable{T,ob}) where {T,ob} = outgoing(parent(sat), ob)
 
-Tables.columnnames(sat::StructACSetTable) = Base.propertynames(sat)
+Tables.columnnames(sat::ACSetTable) = Base.propertynames(sat)
 
-Tables.rowaccess(sat::StructACSetTable) = true
-Tables.rows(sat::StructACSetTable{T,ob}) where {T,ob} =
-  StructACSetRow{T,ob}.(Ref(parent(sat)), parts(parent(sat), ob))
+Tables.rowaccess(sat::ACSetTable) = true
+Tables.rows(sat::ACSetTable{T,ob}) where {T,ob} =
+  ACSetRow{T,ob}.(Ref(parent(sat)), parts(parent(sat), ob))
 
 
-Tables.getcolumn(row::StructACSetRow{T,ob}, i::Int) where {T,ob} =
+Tables.getcolumn(row::ACSetRow{T,ob}, i::Int) where {T,ob} =
   Base.getproperty(row, outgoing(parent(row), ob)[i])
 
-Base.getproperty(row::StructACSetRow, nm::Symbol) = subpart(parent(row), idx(row), nm)
+Base.getproperty(row::ACSetRow, nm::Symbol) = subpart(parent(row), idx(row), nm)
+
+Tables.getcolumn(row::ACSetRow, nm::Symbol) = Base.getproperty(row, nm)
+
+Base.propertynames(row::ACSetRow{T,ob}) where {T,ob} = outgoing(parent(row), ob)
 
-Tables.getcolumn(row::StructACSetRow, nm::Symbol) = Base.getproperty(row, nm)
+Tables.columnnames(row::ACSetRow) = Base.propertynames(row)
 
-Base.propertynames(row::StructACSetRow{T,ob}) where {T,ob} = outgoing(parent(row), ob)
+# ACSet Macro
+#############
 
-Tables.columnnames(row::StructACSetRow) = Base.propertynames(row)
+@ct_enable function _make_acset(@ct(S), T, rows::NamedTuple{names}) where {names}
+  @ct begin
+    s = SchemaDesc(S)
+  end
+  acs = T()
+  @ct_ctrl for ob in intersect(s.obs, names)
+    add_parts!(acs, @ct(ob), rows[@ct ob])
+  end
+  @ct_ctrl for f in intersect([s.homs; s.attrs], names)
+    set_subpart!(acs,:,@ct(f), rows[@ct f])
+  end
+  acs
+end
+
+function make_acset(T::Type{<:StructACSet{S}}, rows::NamedTuple) where {S}
+  _make_acset(Val{S}, T, rows)
+end
+
+macro acset(head, body)
```

Comment:
This macro should have a docstring. I guess it didn't have one before but we may as well add one now.